pdnshttp/client.go

```diff
@@ -19,8 +20,18 @@ type Client struct {
 
 // NewClient returns a new PowerDNS HTTP client
 func NewClient(baseURL string, hc *http.Client, auth ClientAuthenticator, debugOutput io.Writer) *Client {
+	u, err := url.ParseRequestURI(baseURL)
+	if err != nil {
+		panic(err)
+	}
+	var path string
+	if strings.TrimSuffix(u.Path, "/") == "" {
+		path = "/api/v1"
+	} else {
+		path = strings.TrimSuffix(u.Path, "/")
+	}
 	c := Client{
-		baseURL:       baseURL,
+		baseURL:       strings.TrimSuffix(baseURL, "/") + path,
```

Comment:
Careful, if I read this correctly, this would duplicate the path, when it's already contained in the `baseURL`. Might be better to actually _replace_ the path in the parsed URL, and then build a new baseURL from the parsed URL.

```go
u.Path = path
c := Client {
    baseURL: u.String(),
    // ...
}
```